src/api/java/baritone/api/Settings.java

```diff
@@ -1302,6 +1303,13 @@ public final T get() {
             return value;
         }
 
+        public void set(T value) {
+            this.value = value;
+            if (BaritoneAPI.getProvider() != null) {
+                BaritoneAPI.getProvider().getAllBaritones().forEach(iBaritone -> iBaritone.getGameEventHandler().onSettingChanged(new SettingChangedEvent(this)));
+            }
+        }
+
```

Comment:
Not sure about this one.
Simply relying on this method being called means silently breaking compatibility with every single Baritone GUI, causing wrong behavior (ignoring or partially ignoring the changed value) whenever such a GUI is used.
Since we already have an api where others may silently change the settings I think it would be better to periodically check for changes. We can still provide this method and deprecate not using it though (and we can drop the check when this was called since in that case the cache is invalidated anyway).